base/compiler/abstractinterpretation.jl

```diff
@@ -189,7 +200,7 @@ function abstract_call_method_with_const_args(@nospecialize(rettype), @nospecial
             # in this case, see if all of the arguments are constants
             for a in argtypes
                 a = widenconditional(a)
-                if !isa(a, Const) && !isconstType(a)
+                if !isa(a, Const) && !isconstType(a) && !isa(a, PartialStruct)
```

Comment:
Why was this heuristic relaxed? It isn't part of the rest of the PR, so it makes the comment at the top of the loop wrong and should have a separate test added.

Reply:
I don't think we can do the closure case otherwise. I can remove this and change the test to set the inference flag though if you'd prefer.